contracts/commands/AaveV3BasicBuyCommandV2.sol

```diff
@@ -0,0 +1,225 @@
+        validateSelector(operationExecutor.executeOp.selector, executionData);
+        IAccountImplementation(basicBuyTriggerData.positionAddress).execute(
+            address(operationExecutor),
+            executionData
+        );
+    }
+
+    /**
+     *  @inheritdoc ICommand
+     */
+    function isTriggerDataValid(
+        bool omitted,
+        bytes memory triggerData
+    ) external view override returns (bool) {
+        BasicBuyTriggerData memory basicBuyTriggerData = abi.decode(
+            triggerData,
+            (BasicBuyTriggerData)
+        );
+
+        (, , , , uint256 maxLtv, ) = lendingPool.getUserAccountData(
+            basicBuyTriggerData.positionAddress
+        );
+
+        // calcualte +-0.5% deviation bounds from the target LTV
+        (uint256 lowerBoundTarget, uint256 upperBoundTarget) = basicBuyTriggerData.targetLTV.bounds(
+            basicBuyTriggerData.deviation
+        );
+
+        // assure that the execution LTV is lower than the lower target, so it wont execute again
+        bool executionLtvBelowLowerTarget = basicBuyTriggerData.executionLTV < lowerBoundTarget;
+        // assure that the trigger type is the correct one
+        bool triggerTypeCorrect = basicBuyTriggerData.triggerType == AAVE_V3_BASIC_BUY_TRIGGER_TYPE;
+        // assure that the upper bound of target LTV is lower than the max LTV, it would revert on execution
+        bool upperTargetBelowMaxLtv = upperBoundTarget < maxLtv;
+        // assure that the deviation is valid ( above minimal allowed deviation)
+        bool deviationValid = deviationIsValid(basicBuyTriggerData.deviation);
+        return
+            executionLtvBelowLowerTarget &&
+            triggerTypeCorrect &&
+            upperTargetBelowMaxLtv &&
+            deviationValid;
+    }
+
+    /**
+     *  @inheritdoc ICommand
+     */
+    function isExecutionLegal(bytes memory triggerData) external view override returns (bool) {
+        BasicBuyTriggerData memory basicBuyTriggerData = abi.decode(
+            triggerData,
+            (BasicBuyTriggerData)
+        );
+        /* 
+        totalCollateralBase - total collateral of the user, in market’s base currency
+        totalDebtBase - total debt of the user, in market’s base currency
+        maxLtv - maximum Loan To Value of the user - weighted average of max ltv of collateral reserves */
+        (uint256 totalCollateralBase, uint256 totalDebtBase, , , uint256 maxLtv, ) = lendingPool
+            .getUserAccountData(basicBuyTriggerData.positionAddress);
+
+        if (totalCollateralBase == 0 || totalDebtBase == 0) {
+            return false;
+        }
+
+        /* Calculate the loan-to-value (LTV) ratio for Aave V3
+         LTV is the ratio of the total debt to the total collateral, expressed as a percentage
+         The result is multiplied by 10000 to preserve precision
+         eg 0.67 (67%) LTV is stored as 6700 */
+        uint256 ltv = (totalDebtBase * 10000) / totalCollateralBase;
```

Comment:
Suggestion: If some of these calculations are common (I see them in both buy and sell), we could create a library with all internal functions (so it gets included in the contract), that can be used by both contracts and if we need to update them it can be done in a single place